include/RAJA/policy/hip/intrinsics.hpp

```diff
@@ -233,18 +233,18 @@ RAJA_DEVICE RAJA_INLINE T warp_reduce(T val, T RAJA_UNUSED_ARG(identity))
 
   T temp = val;
 
-  if (numThreads % policy::hip::WARP_SIZE == 0) {
+  if (numThreads % policy::hip::device_constants.WARP_SIZE == 0) {
```

Comment:
A lot of stuff in this file looks to me like a verbatim copy of what's in the similarly named CUDA file? Is it worth consolidating?

Reply:
The entire cuda and hip backends are kept as practically verbatim copies. Why stop now?

Reply:
Agreed. Perhaps something to think about addressing in the future when we have time. 😄